src/api.v2/helpers/access/createUserInvite.js

```diff
@@ -0,0 +1,43 @@
+const UserAccess = require('../../model/UserAccess');
+
+const { getAwsUserAttributesByEmail } = require('../../../utils/aws/user');
+const sendEmail = require('../../../utils/send-email');
+const buildUserInvitedEmailBody = require('../../../utils/emailTemplates/buildUserInvitedEmailBody');
+const buildUserInvitedNotRegisteredEmailBody = require('../../../utils/emailTemplates/buildUserInvitedNotRegisteredEmailBody');
+
+const OK = require('../../../utils/responses/OK');
+
+const getLogger = require('../../../utils/getLogger');
+
+const logger = getLogger('[AccessModel] - ');
+
+const createUserInvite = async (experimentId, userEmail, role, inviterUser) => {
+  let userAttributes;
+  try {
+    userAttributes = await getAwsUserAttributesByEmail(userEmail);
+  } catch (e) {
+    if (e.code !== 'UserNotFoundException') {
+      throw e;
+    }
+
+    logger.log('User has not yet signed up, inviting new user');
+  }
+
+
+  let emailBody;
+  if (!userAttributes) {
+    new UserAccess().addToInviteAccess(userEmail, experimentId, role);
+    emailBody = buildUserInvitedNotRegisteredEmailBody(userEmail, inviterUser);
```

Comment:
You can have this part in the catch after the check `e.code !== 'UserNotFoundException'`